crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -36,24 +36,68 @@ fn ambiguous_with(graph_info: &mut GraphInfo, set: InternedSystemSet) {
     }
 }
 
+/// Stores data to differentiate different Node types
+pub trait NodeType {
+    /// Used in `NodeConfig`
+    type Metadata;
+    /// Used in `NodeConfigs`
+    type GroupMetadata;
+
+    /// Initialize `NodeConfig` from `NodeType`
+    fn config(self) -> NodeConfig<Self>
+    where
+        Self: Sized;
+}
+
+impl NodeType for ScheduleSystem {
+    type Metadata = GraphInfo;
+    type GroupMetadata = Chain;
+
+    fn config(self) -> NodeConfig<Self> {
+        let sets = self.default_system_sets().into_iter().collect();
```

Comment:
```suggestion
        let sets = self.default_system_sets().clone();
```
This is actually something I noticed a while ago but haven't gotten around to changing, lol.